modules/wallet/persist.go

```diff
@@ -37,8 +38,33 @@ type WalletSettings struct {
 	// 32 '0' bytes.
 	EncryptionVerification crypto.Ciphertext
 
-	PrimarySeed     string // Name of the primary seed for the wallet.
-	AddressProgress uint64 // Number of addresses used in the primary seed.
+	// The primary seed is used to generate new addresses as they are required.
+	// All addresses are tracked and spendable. Only modules.PublicKeysPerSeed
+	// keys/addresses can be created per seed, after which a new seed will need
+	// to be generated.
+	PrimarySeedFile     SeedFile
+	PrimarySeedProgress uint64
+	PrimarySeedFilename string
+}
+
+// unlockKey creates a wallet unlocking key from the input master key.
+func unlockKey(masterKey crypto.TwofishKey) crypto.TwofishKey {
+	keyBase := append(masterKey[:], unlockModifier[:]...)
+	return crypto.TwofishKey(crypto.HashObject(keyBase))
+}
+
+// checkMasterKey verifies that the master key is correct.
+func (w *Wallet) checkMasterKey(masterKey crypto.TwofishKey) error {
+	unlockKey := unlockKey(masterKey)
+	verification, err := unlockKey.DecryptBytes(w.settings.EncryptionVerification)
+	if err != nil {
+		return err
+	}
+	expected := make([]byte, encryptionVerificationLen)
+	if !bytes.Equal(expected, verification) {
```

Comment:
Comparing to zero confused me for a bit. I thought it might be an error. Perhaps a comment would be helpful here. Better would be a `bytes.IsEmpty` function, but it doesn't exist.